Services/data/devicemodels/scripts/chiller-02-state.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+/*global log*/
+/*global updateState*/
+/*global updateProperty*/
+/*global sleep*/
+/*jslint node: true*/
+
+"use strict";
+
+// Default state
+var state = {
+    online: true,
+    temperature: 75.0,
+    temperature_unit: "F",
+    humidity: 70.0,
+    humidity_unit: "%",
+    pressure: 250.0,
+    pressure_unit: "psig",
+    simulation_state: "high_pressure"
+};
+
+// Default device properties
+var properties = {};
+
+/**
+ * Restore the global state using data from the previous iteration.
+ *
+ * @param previousState device state from the previous iteration
+ * @param previousProperties device properties from the previous iteration
+ */
+function restoreSimulation(previousState, previousProperties) {
+    // If the previous state is null, force a default state
+    if (previousState) {
+        state = previousState;
+    } else {
+        log("Using default state");
+    }
+
+    if (previousProperties) {
+        properties = previousProperties;
+    } else {
+        log("Using default properties");
+    }
+}
+
+/**
+ * Simple formula generating a random value around the average
+ * in between min and max
+ *
+ * @returns random value with given parameters
+ */
+function vary(avg, percentage, min, max) {
+    var value = avg * (1 + ((percentage / 100) * (2 * Math.random() - 1)));
+    value = Math.max(value, min);
+    value = Math.min(value, max);
+    return value;
+}
+
+/**
+ * Entry point function called by the simulation engine.
+ * Returns updated simulation state.
+ * Device property updates must call updateProperties() to persist.
+ *
+ * @param context             The context contains current time, device model and id
+ * @param previousState       The device state since the last iteration
+ * @param previousProperties  The device properties since the last iteration
+ */
+/*jslint unparam: true*/
+function main(context, previousState, previousProperties) {
```

Comment:
could you please add unit tests for these and other files too.

Reply:
Thanks for pointing it out, forgot to add it for new files. Done now.